llvm/lib/Target/EVM/EVMAsmPrinter.cpp

```diff
@@ -105,6 +125,49 @@ bool EVMAsmPrinter::isBlockOnlyReachableByFallthrough(
   return false;
 }
 
+void EVMAsmPrinter::emitLinkerSymbol(const MachineInstr *MI) {
+  MCSymbol *LinkerSymbol = MI->getOperand(0).getMCSymbol();
+  StringRef LinkerSymbolName = LinkerSymbol->getName();
+  unsigned Opc = MI->getOpcode();
+
+  std::string SymbolNameHash = EVM::getLinkerSymbolHash(LinkerSymbolName);
+  std::string DataSymbolNameHash =
+      (Opc == EVM::DATASIZE_S) ? EVM::getDataSizeSymbol(SymbolNameHash)
+                               : EVM::getDataOffsetSymbol(SymbolNameHash);
+
+  MCInst MCI;
+  if (Opc == EVM::DATASIZE_S || Opc == EVM::DATAOFFSET_S)
```

Comment:
Can we add an assert in the beginning of the function that these two opcodes are expected? If that is possible, we don't need this check here.

Reply:
Indeed it seems that 
`DataSymbolNameHash` definition uses this assumption anyway.

Reply:
Thank you! I missed to remove this.